### PR TITLE
add probabilistic forecasts to APISession

### LINKED ISSUE #154: Add Probabilistic Forecasts to api.py
Related/ Dependent on to #47 
Functions for retrieving and posting Probabilistic Forecasts to the api will need to be added once the data model objects are completed.

### PR DESCRIPTION
**Pull request by wholmgren**
- [x] Closes #154  .
  - [x] I am familiar with the [contributing guidelines](https://solarforecastarbiter-core.readthedocs.io/en/latest/contributing.html).
  - [x] Tests added.
  - [x] Updates entries to [`docs/source/api.rst`](https://github.com/SolarArbiter/solarforecastarbiter-core/blob/master/docs/source/api.rst) for API changes.
  - [x] Adds descriptions to appropriate "what's new" file in [`docs/source/whatsnew`](https://github.com/SolarArbiter/solarforecastarbiter-core/tree/master/docs/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
  - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
  - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

@lboeman see the note in `get_probabilistic_forecast_constant_value` regarding https://github.com/SolarArbiter/solarforecastarbiter-api/issues/158

### CODE REVIEW

**Comment by wholmgren** [comment]
@alorenzo175 this is ready for review. I'll implement the real API session tests after your feedback.

File: solarforecastarbiter/io/api.py [hunk 1]
```diff
@@ -292,6 +292,112 @@ def create_forecast(self, forecast):
...
+
+        Returns
+        -------
+        datamodel.ProbabilisticForecastConstantValue
+        """
+        # add /metadata after
+        # https://github.com/SolarArbiter/solarforecastarbiter-api/issues/158
+        req = self.get(f'/forecasts/cdf/single/{forecast_id}')
+        fx_dict = req.json()
+        site = self.get_site(fx_dict['site_id'])
```
[the remark this answers is not in the document]
Comment (wholmgren) [reply solarforecastarbiter/io/api.py]: good idea, though I'm not sure about the handling for `site.site_id !=  fx_dict['site_id']`. Perhaps better to raise a ValueError if it's not the same?

[the remark this answers is not in the document]
Comment (alorenzo175) [reply solarforecastarbiter/io/api.py]: yeah, probably better

**Comment by wholmgren** [comment]
@alorenzo175 added the real session tests, but not sure what to do about the database ids. For now, I've picked something called [DA GHI](https://dashboard.solarforecastarbiter.org/forecasts/cdf/ef51e87c-50b9-11e9-8647-d663bd873d93), but I'm guessing that's not a long term object. Also, `test_real_apisession_post_prob_forecast_constant_val_values` appears to need something to point at because it doesn't clean up after itself (following the pattern for the other tests).

File: solarforecastarbiter/io/api.py [hunk 2]
```diff
@@ -292,6 +292,123 @@ def create_forecast(self, forecast):
...
+            return []
+        sites = {site.site_id: site for site in self.list_sites()}
+        out = []
+        for fx_dict in fx_dicts:
+            site = sites.get(fx_dict['site_id'])
+            fx_dict['site'] = site
+            cvs = []
+            for constant_value_dict in fx_dict['constant_values']:
+                cvs.append(self.get_probabilistic_forecast_constant_value(
+                    constant_value_dict['forecast_id']))
```
Comment (alorenzo175) [inline solarforecastarbiter/io/api.py]: site=site

File: solarforecastarbiter/io/api.py [hunk 3]
```diff
@@ -292,6 +292,123 @@ def create_forecast(self, forecast):
...
+        forecast_id : string
+            UUID of the forecast to get metadata for
+        site : datamodel.Site or None
+            If provided, the object will be attached to the returned
+            value (faster). If None, object will be created from site
+            metadata obtained from the database (slower).
+
+        Returns
+        -------
+        datamodel.ProbabilisticForecastConstantValue
```
Comment (alorenzo175) [inline solarforecastarbiter/io/api.py]: raises valueerror

**Comment by alorenzo175** [comment]
> @alorenzo175 added the real session tests, but not sure what to do about the database ids. For now, I've picked something called [DA GHI](https://dashboard.solarforecastarbiter.org/forecasts/cdf/ef51e87c-50b9-11e9-8647-d663bd873d93), but I'm guessing that's not a long term object. Also, `test_real_apisession_post_prob_forecast_constant_val_values` appears to need something to point at because it doesn't clean up after itself (following the pattern for the other tests).
 
DA GHI should be fine and I think initiated by the database migration. Why not use the same constant value id as in the get test?

**Comment by wholmgren** [comment]
Works for me if it's part of the database migration. I think this is ready to merge now.

**Pull request merged**